base/gmp.jl

```diff
@@ -41,11 +41,30 @@ type BigInt <: Integer
     size::Cint
     d::Ptr{Limb}
     function BigInt()
+        if length(bigintpool) > 0
+            b = pop!(bigintpool)
+            finalizer(b, poolingfinalizer)
+            #warn("\n\nREUSING!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!\n\n")
+            #@show b
+            return b
+        end
         b = new(zero(Cint), zero(Cint), C_NULL)
         ccall((:__gmpz_init,:libgmp), Void, (Ptr{BigInt},), &b)
-        finalizer(b, _gmp_clear_func)
+        finalizer(b, poolingfinalizer)
         return b
```

Comment:
would probably be better style to have
```
         if isempty(bigintpool)
             b = new(zero(Cint), zero(Cint), C_NULL)
             ccall((:__gmpz_init,:libgmp), Void, (Ptr{BigInt},), &b)
         else
             b = pop!(bigintpool)
         endif
         finalizer(b, poolingfinalizer)
         return b
```
so that the `finalizer`/`return` lines appear only once.